src/renderer/widgets/CreateWallet/model/form-model.ts

```diff
@@ -0,0 +1,113 @@
+import { accountUtils, walletModel, walletUtils } from '@entities/wallet';
+import { networkUtils } from '@entities/network';
+import { FormParams } from '../lib/types';
+import { signatoryModel } from './signatory-model';
+import { toAccountId } from '@shared/lib/utils';
+
+const $createMultisigForm = createForm<FormParams>({
+  fields: {
+    threshold: {
+      init: 0,
+      rules: [
+        {
+          name: 'moreOrEqualToTwo',
+          validator: (threshold) => threshold >= 2,
+        },
+      ],
+    },
+    chain: {
+      init: chains[0] as Chain,
+    },
+    name: {
+      init: '',
+      rules: [
+        {
+          name: 'notEmpty',
+          validator: (name) => name !== '',
+          errorText: 'createMultisigAccount.disabledError.emptyName',
+        },
+      ],
+    },
+  },
+  validateOn: ['submit'],
+});
+
+const formSubmitted = createEvent<void>();
+
+const $multisigAccountId = combine(
+  {
+    formValues: $createMultisigForm.$values,
+    signatories: signatoryModel.$signatories,
+  },
+  ({ formValues: { threshold, chain }, signatories }) => {
+    if (!threshold) return null;
+
+    const cryptoType = networkUtils.isEthereumBased(chain.options) ? CryptoType.ETHEREUM : CryptoType.SR25519;
+
+    return accountUtils.getMultisigAccountId(
+      Array.from(signatories.values()).map((s) => toAccountId(s.address)),
+      threshold,
+      cryptoType,
+    );
+  },
+);
+
+const $multisigAlreadyExists = combine(
+  { wallets: walletModel.$wallets, multisigAccountId: $multisigAccountId, formValues: $createMultisigForm.$values },
+  ({ multisigAccountId, wallets, formValues: { chain } }) =>
+    !!walletUtils.getWalletFilteredAccounts(wallets, {
+      walletFn: (w) => walletUtils.isMultisig(w),
+      accountFn: (multisigAccount) => {
+        const isSameAccountId = multisigAccount.accountId === multisigAccountId;
+        const chainId = (multisigAccount as MultisigAccount).chainId;
+        const isSameChainId = !chainId || chainId === chain.chainId;
+
+        return isSameAccountId && isSameChainId;
+      },
+    }),
+  { skipVoid: false },
+);
+
+const $availableAccounts = combine(
+  {
+    wallets: walletModel.$wallets,
+    formValues: $createMultisigForm.$values,
+  },
+  ({ formValues: { chain }, wallets }) => {
+    if (!chain) return [];
+
+    const filteredAccounts = walletUtils.getAccountsBy(wallets, (a, w) => {
+      const isValidWallet = !walletUtils.isWatchOnly(w) && !walletUtils.isProxied(w) && !walletUtils.isMultisig(w);
+      const isChainMatch = accountUtils.isChainAndCryptoMatch(a, chain);
+
+      return isValidWallet && isChainMatch;
+    });
+
+    const baseAccounts = filteredAccounts.filter((a) => accountUtils.isBaseAccount(a) && a.name);
+
+    return [...filteredAccounts, ...baseAccounts];
+  },
+);
+
+sample({
+  clock: $createMultisigForm.submit,
+  source: $createMultisigForm.$isValid,
+  filter: (isValid: boolean) => isValid,
```

Comment:
I assume `boolean` is not required

Reply:
I get an error with isValid being implicitely of type any, although `$createMultisigForm.$isValid` is of type `StoreWritable<boolean>`. Maybe something specific with effector-forms?

Reply:
Alright, let's leave it as is